src/hub.js

```diff
@@ -352,14 +370,31 @@ const onReady = async () => {
   const initialEnvironmentEl = document.createElement("a-entity");
   initialEnvironmentEl.addEventListener("bundleloaded", () => {
     remountUI({ initialEnvironmentLoaded: true });
-    // Wait a tick plus some margin so that the environments actually render.
-    setTimeout(() => scene.renderer.animate(null), 100);
+    // We never want to stop the render loop when were running in "bot" mode.
+    if (!qsTruthy("bot")) {
+      // Stop rendering while the UI is up. We restart the render loop in enterScene.
+      // Wait a tick plus some margin so that the environments actually render.
+      setTimeout(() => scene.renderer.animate(null), 100);
+    }
   });
   environmentRoot.appendChild(initialEnvironmentEl);
 
+  const setRoom = (janusRoomId, hubName) => {
+    if (qsTruthy("bot")) {
+      const enterSceneImmediately = () => enterScene(new MediaStream(), false, janusRoomId);
+      if (scene.hasLoaded) {
+        enterSceneImmediately();
+      } else {
+        scene.addEventListener("loaded", enterSceneImmediately);
+      }
+    } else {
+      remountUI({ janusRoomId, hubName });
+    }
+  };
+
   if (qs.room) {
     // If ?room is set, this is `yarn start`, so just use a default environment and query string room.
-    remountUI({ janusRoomId: qs.room && !isNaN(parseInt(qs.room)) ? parseInt(qs.room) : 1 });
+    setRoom(qs.room && !isNaN(parseInt(qs.room)) ? parseInt(qs.room) : 1);
```

Comment:
I think this inadvertently will set `hubName` on the UI props to `undefined`, since it's not passed in, right?

Reply:
(Probably nbd but just thought I'd mention it)

Reply:
Yeah, this happens when you use `hub.html?room=...` anyway.